tests/multicomponent_compressible_projected_density_box.prm

```diff
@@ -0,0 +1,165 @@
+# This is a testcase that demonstrates the projected density approximation within 
+# Multicomponent Compressible. 
+# The setup is a 1x1 km, 2x2 grid where we apply a surface pressure through time. 
+# This allows for no material to flow into/out of the model while allowing 
+# density to accomodate the increased pressure.
+# 
+# The equation being solved is from the multicomponent compressible equation of state
+
+# reference_thermal_expansivities              = 2e-5
+# reference_isothermal_compressibilities       = 4e-12
+# temperature                                  = 1600
+# reference_temperatures                       = 1600
+# isothermal_bulk_modulus_pressure_derivatives = 4
+# reference_densities                          = 3300
+# pressure                                     = 5e10
+
+# ak = reference_thermal_expansivities[c]/reference_isothermal_compressibilities[c]
+# f  = (1. + (pressure - ak*(temperature - reference_temperatures[c])) *
+#                    isothermal_bulk_modulus_pressure_derivatives[c]   *
+#                          reference_isothermal_compressibilities[c])
+# densities[c] = reference_densities[c]*std::pow(f, 1./isothermal_bulk_modulus_pressure_derivatives[c])
+
+# The seperately calculated solution for density is 3822.3642114512877 and as we are only
+# using one field, the averaging done by the material model is between only the background and our
+# density field, both of which have a density of 3300.
+
+set Dimension = 2 
+set End time                               = 1e6
+set Use years in output instead of seconds = true
+set Adiabatic surface temperature          = 1600
+set Surface pressure                       = 5e10
+set Pressure normalization                 = surface
+set Nonlinear solver scheme                = iterated Advection and Stokes
+set Nonlinear solver tolerance             = 1e-12
+set Max nonlinear iterations               = 25
+set Output directory                       = visco_plastic_compressible_box
+
+# This calculates the adiabatic temperature profile down the model and is used
+# by the initial temperature and composition models.
+subsection Adiabatic conditions model
+  subsection Compute profile
+    set Number of points              = 100
+    set Composition reference profile = function
+  end
+end
+
+subsection Solver parameters
+  subsection Stokes solver parameters
+    set Linear solver tolerance = 1e-12
+  end
+end
+
+# As the model is testing compressibility we want to use a locally conservative 
+# Stokes discretization, this causes some discontinuities in pressure in the
+# finite element space between cells. Being locally conservative means that 
+# the inflow and outflow of a given cell are blanced. We are not advecting
```

Comment:
Does the test not work if the discretization is continuous? If it doesn't ignore this comment. But you might be able to delete this section